File: pset6/readability/readability.py
from cs50 import get_string

# counters for letters, sentences, and words
letters = 0
sentences = 0
words = 1

# get user input
text = get_string("Text: ")

# count letters
for x in text:
    if (ord(x) >= 65 and ord(x)) or (ord(x) >= 97 and ord(x) <= 122):
        letters += 1

# count words
for x in text:
    if ord(x) == 32:
        words += 1

# count sentences
for x in text:
    if ord(x) == 33 or ord(x) == 46 or ord(x) == 63:
        sentences += 1

# calculate and print grade level
L = letters * 100 / words
S = sentences * 100 / words
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Counter for letters, sentences, and words
    float letters = 0;
    float sentences = 0;
    float words = 1;

    // Get input text and create variable for text length
    string text = get_string("Text: ");
    int length = strlen(text);

    // Count how many letters
    for (int i = 0; i < length; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters++;
        }
    }

    // Count how many words
    for (int i = 0; i < length; i++)
    {
        if (text[i] == 32)
        {
            words++;
        }
    }

    // Count how many sentences
    for (int i = 0; i < length; i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
    }

    // Calculate and print grade level
    float L = letters * 100 / words;
    float S = sentences * 100 / words;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
       printf("Grade %i\n", index);
    }
}